docs/en/modules/gpio.md

```diff
@@ -98,18 +98,18 @@ gpio.serout(1,1,{8,18},8) -- serial 30% pwm 38k, lasts 8 cycles
 
 ## gpio.trig()
 
-Establish a callback function to run on interrupt for a pin.
-
-There is currently no support for unregistering the callback.
+Establish or clear a callback function to run on interrupt for a pin.
 
 This function is not available if GPIO_INTERRUPT_ENABLE was undefined at compile time.
 
 #### Syntax
-`gpio.trig(pin, type [, function(level)])`
+`gpio.trig(pin, [type [, function(level)]})`
 
 #### Parameters
 - `pin` **1~12**, IO index, pin D0 does not support interrupt.
-- `type` "up", "down", "both", "low", "high", which represent rising edge, falling edge, both edge, low level, high level trig mode correspondingly.
+- `type` "up", "down", "both", "low", "high", which represent rising edge, falling edge, both edge, 
+low level, high level trig mode correspondingly. If the type is "none" or omitted then the
+callback function is removed and the interrupt disabled.
 - `function(level)` callback function when triggered. The gpio level is the param. Use previous callback function if undefined here.
```

Comment:
Are you looking at the same code as I submitted on this patch because it had been updated an hour or so before your comment? 

A value of "none" or omitted sets the type = `GPIO_PIN_INTR_DISABLE`, so the branch validating the callback function isn't taken and the cb ref is set to `LUA_NOREF`. I did try the test cases of setting a trigger, and mutilple times  and deleting the trigger with the ommitted and "none" parameters and they work fine as per the documentation. I don't understand your comment.

 Any old registry entry is unref'ed to allow correct GC to take place.